src/boot/efi/linux_x86.c

```diff
@@ -153,12 +152,25 @@ EFI_STATUS linux_exec_efi_handover(
         bool can_4g = image_params->hdr.version >= SETUP_VERSION_2_12 &&
                         FLAGS_SET(image_params->hdr.xloadflags, XLF_CAN_BE_LOADED_ABOVE_4G);
 
-        if (!can_4g && POINTER_TO_PHYSICAL_ADDRESS(linux_buffer) + linux_length > UINT32_MAX)
-                return log_error_status(
-                                EFI_UNSUPPORTED,
-                                "Unified kernel image was loaded above 4G, but kernel lacks support.");
-        if (!can_4g && POINTER_TO_PHYSICAL_ADDRESS(initrd_buffer) + initrd_length > UINT32_MAX)
-                return log_error_status(EFI_UNSUPPORTED, "Initrd is above 4G, but kernel lacks support.");
+        /* There is no way to pass the high bits of code32_start. Newer kernels seems to handle this
+         * just fine, but older kernels will fail even if they otherwise have above 4G boot support. */
```

Comment:
Do we know which kernel version fixed this? So that we know when the workaround can be dropped

Reply:
I believe any that contain https://github.com/torvalds/linux/commit/04a7d0e15606769ef58d5cee912c5d08d93ded92. Which is to say, any such kernel should not be going through the EFI handover code at all. So I wouldn't worry and just keep it until handover support is dropped.